app/buckets/handlers.js

```diff
@@ -37,11 +37,24 @@ exports.create_bucket = (req, res) => {
 exports.bucket_details = (req, res, next) => {
   Promise.all([Bucket.get(req.params.id), App.list(), User.list()])
     .then(([bucket, apps, users]) => {
+      const current_user_is_bucket_admin = bucket.has_admin(req.user.auth0_id);
       res.render('buckets/details.html', {
         bucket,
         apps_options: apps.exclude(bucket.apps),
         users_options: users.exclude(bucket.users),
+        current_user_is_bucket_admin
       });
     })
     .catch(next);
 };
+
+exports.delete = (req, res, next) => {
+  Promise.all([Bucket.get(req.params.id), Bucket.delete(req.params.id)])
```

Comment:
Doing these in parallel causes a race condition. What if `delete` happens before `get`? Either chain these, eg:
```javascript
Bucket.get(req.params.id)
  .then((bucket) => {
    return Bucket.delete(bucket.id);
  })
  .then(() => {
    ...redirect...
  })
  .catch(next);
```
or, if you change the flash message to not rely on `bucket.name`, you can just do `delete` without the `get`.

Reply:
True, but I only did the `.get` because I wanted the bucket name. It's nicer. I'll chain them. 👍